core/vdbe/mod.rs

```diff
@@ -238,14 +238,37 @@ fn get_cursor_as_sorter_mut<'long, 'short>(
     cursor
 }
 
+struct Bitfield<const N: usize>([u64; N]);
+
+impl<const N: usize> Bitfield<N> {
+    fn new() -> Self {
+        Self([0; N])
+    }
+
+    fn set(&mut self, bit: usize) {
+        assert!(bit < N * 64, "bit out of bounds");
+        self.0[bit / 64] |= 1 << (bit % 64);
+    }
+
+    fn unset(&mut self, bit: usize) {
+        assert!(bit < N * 64, "bit out of bounds");
+        self.0[bit / 64] &= !(1 << (bit % 64));
+    }
+
+    fn get(&self, bit: usize) -> bool {
+        assert!(bit < N * 64, "bit out of bounds");
+        (self.0[bit / 64] & (1 << (bit % 64))) != 0
+    }
+}
+
 /// The program state describes the environment in which the program executes.
 pub struct ProgramState {
     pub pc: InsnReference,
     cursors: RefCell<BTreeMap<CursorID, Cursor>>,
     registers: Vec<OwnedValue>,
     last_compare: Option<std::cmp::Ordering>,
     deferred_seek: Option<(CursorID, CursorID)>,
-    ended_coroutine: HashMap<usize, bool>, // flag to indicate that a coroutine has ended (key is the yield register)
+    ended_coroutine: Bitfield<4>, // flag to indicate that a coroutine has ended (key is the yield register. currently we assume that the yield register is always between 0-255, YOLO)
```

Comment:
using this improves a few nanoseconds in `SELECT 1`, probably not worth it

Reply:
mainly i just wanted to use const generics

Reply:
Every nanosecond counts